modules/extension/app-schema/app-schema/src/main/java/org/geotools/jdbc/NamespaceAwareAttributeRenameVisitor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2016, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.jdbc;
+
+import org.geotools.filter.NestedAttributeExpression;
+import org.geotools.filter.visitor.DuplicatingFilterVisitor;
+import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.PropertyName;
+
+/**
+ * Renames the specified attribute to a new target name, preserving the namespace context.
+ * 
+ * @author Stefano Costa, GeoSolutions
+ *
+ */
+public class NamespaceAwareAttributeRenameVisitor extends DuplicatingFilterVisitor {
+
+    String sourceProperty;
+
+    String targetProperty;
+
+    public NamespaceAwareAttributeRenameVisitor(String sourceProperty, String targetProperty) {
+        super();
+        this.sourceProperty = sourceProperty;
+        this.targetProperty = targetProperty;
+    }
+
+    public Expression visit(NestedAttributeExpression expression, Object extraData) {
```

Comment:
These two ```visit``` methods should have ```@Override``` or javadocs. If these are overrides, convention is that docs can be found in the parent class or interface.